cdap-app-fabric/src/main/java/co/cask/cdap/app/DefaultApplicationContext.java

```diff
@@ -26,8 +30,25 @@
  */
 public class DefaultApplicationContext<T extends Config> implements ApplicationContext<T> {
 
+  private static final Gson GSON = new Gson();
+  private final T appConfig;
+
+  public DefaultApplicationContext() {
+    this.appConfig = null;
+  }
+
+  //TODO : Handle JsonSyntax Exception better and throw a more meaningful error. This will be done when we
+  // use PluginInstantiator methods instead of GSON deserialization
+  public DefaultApplicationContext(TypeToken<?> configToken, @Nullable String configString) {
```

Comment:
think it would be better to take the Type rather than the TypeToken. Then gson is an internal detail and I can just pass in MyConfig.class directly instead of wrapping it in a type token.